executor/src/witgen/jit/symbolic_expression.rs

```diff
@@ -279,7 +279,10 @@ impl<T: FieldElement, V: Clone> SymbolicExpression<T, V> {
 
     /// Integer division, i.e. convert field elements to unsigned integer and divide.
     pub fn integer_div(&self, rhs: &Self) -> Self {
-        if rhs.is_known_one() {
+        if let (SymbolicExpression::Concrete(a), SymbolicExpression::Concrete(b)) = (self, rhs) {
+            assert!(b != &T::from(0));
+            SymbolicExpression::Concrete(*a / *b)
+        } else if rhs.is_known_one() {
```

Comment:
We should maybe also add `is_known_minus_one`

Reply:
And `self.is_known_zero`

Reply:
Sure, although it's a lot less important, no? I would expect the Rust optimizer to do that for us. Whereas the case I added makes the difference between a concrete value and a known but in-concrete value.

Reply:
Actually, if you agree I'd leave it as is for this reason, just to not clutter it too much.

Also, it looks like for `field_div` we currently do
```rs
if self.is_known_zero() {
    SymbolicExpression::Concrete(T::from(0))
}
```

But what if `rhs` is zero? Maybe it would be better to remove this?

Reply:
I think the assumption was that `field_div` is always preceded by an assertion.

Reply:
`unchecked_field_div`?

Reply:
It's release code, so we can assume all checks have been removed  ;)

Reply:
Oh I just read the comment 10 lines up, it does say that it requires a check for division by zero at the outside.